ext/spl/spl_observer.c

```diff
@@ -361,6 +443,69 @@ PHP_METHOD(SplObjectStorage, attach)
 	spl_object_storage_attach(intern, obj, inf);
 } /* }}} */
 
+static int spl_object_storage_has_dimension(zend_object *object, zval *offset, int check_empty)
+{
+	spl_SplObjectStorage *intern = spl_object_storage_from_obj(object);
+	if (UNEXPECTED(offset == NULL || Z_TYPE_P(offset) != IS_OBJECT || (intern->flags & SOS_OVERRIDDEN_READ_DIMENSION))) {
```

Comment:
Can we have an overriden ``offsetExists()``?

Reply:
SOS_OVERRIDDEN_READ_DIMENSION includes fptr_gethash, offsetget, and offsetexists checks